app/components/pay-page/choose-membership-plan-modal.ts

```diff
@@ -0,0 +1,108 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import fade from 'ember-animated/transitions/fade';
+import { inject as service } from '@ember/service';
+import Store from '@ember-data/store';
+import window from 'ember-window-mock';
+import * as Sentry from '@sentry/ember';
+import type RegionalDiscountModel from 'codecrafters-frontend/models/regional-discount';
+import type PromotionalDiscountModel from 'codecrafters-frontend/models/promotional-discount';
+import type AuthenticatorService from 'codecrafters-frontend/services/authenticator';
+
+export interface PricingPlan {
+  id: 'quarterly' | 'yearly' | 'lifetime';
+  title: string;
+  priceInDollars: number;
+  validityInMonths?: number; // undefined for lifetime
+}
+
+export const PRICING_PLANS: PricingPlan[] = [
+  { id: 'quarterly', title: '3 month pass', priceInDollars: 120, validityInMonths: 3 },
+  { id: 'yearly', title: '1 year pass', priceInDollars: 360, validityInMonths: 12 },
+  { id: 'lifetime', title: 'Lifetime pass', priceInDollars: 1490 },
+];
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    activeDiscountForYearlyPlan?: PromotionalDiscountModel | null;
+    onClose: () => void;
+    regionalDiscount?: RegionalDiscountModel | null;
+  };
+}
+
+export default class ChooseMembershipPlanModal extends Component<Signature> {
+  @service declare authenticator: AuthenticatorService;
+  @service declare store: Store;
+
+  transition = fade;
+  pricingPlans = PRICING_PLANS;
+
+  @tracked extraInvoiceDetailsRequested = false;
+  @tracked isCreatingCheckoutSession = false;
+  @tracked currentStep: 'plan-selection' | 'invoice-details' = 'plan-selection';
+  @tracked selectedPlanId: PricingPlan['id'] = 'quarterly';
+
+  get selectedPlan(): PricingPlan {
+    const plan = this.pricingPlans.find((p) => p.id === this.selectedPlanId);
+
+    if (!plan) {
+      Sentry.captureException(new Error(`Unknown selectedPlanId: ${this.selectedPlanId}`));
+
+      return this.pricingPlans[0]!;
+    }
+
+    return plan;
+  }
+
+  @action
+  handleBackButtonClick() {
+    this.currentStep = 'plan-selection';
+  }
+
+  @action
+  handleContinueButtonClick() {
+    this.currentStep = 'invoice-details';
+  }
+
+  @action
+  handleExtraInvoiceDetailsToggle(value: boolean) {
+    this.extraInvoiceDetailsRequested = value;
+  }
+
+  @action
+  handlePlanSelect(plan: PricingPlan) {
+    this.selectedPlanId = plan.id;
+  }
+
+  @action
+  async handleProceedToCheckoutClick() {
+    if (!this.authenticator.isAuthenticated) {
```

Comment:
Let's do this on when a person chooses a plan instead of when they to the "invoice details" thing (that way they only lose one step worth of progress when they hit the page again, doesnt' seem like they filled in a bunch of stuff and then lost context)